### PR TITLE
Add converting auth util functions

### LINKED ISSUE #63: Create More Auth Utility functions to convert between email and UID
### Issue summary

Utility functions need to be made that allow any user's UID to be retrieved given their email and vice versa. This is to be used when storing collaborators for a trip, so that users can input emails for collaborators and developers can convert these to UIDs to store instead. Then when retrieving a trip, the UIDs can be converted back to emails. The purpose of this is to prevent a change of email from locking a user out of their trips. This will likely need to use the Firebase Admin SDK and *possibly* be implemented in a Java servlet.

The specific names of the functions are:
- `getUserEmailArrFromUserUidArr(uid)`
- `getUserUidArrFromUserEmailArr(email)`

### PR DESCRIPTION
**Pull request by keiffer01**
### What is a quick description of the change?

Adds util functions to the frontend that accesses the backend servlets from #99 to convert between user UID and email.

### Is this fixing an issue?

Closes #63.

### Are there more details that are relevant?

Two async functions were added to the `index.js` file of the `AuthUtils` directory: `getUserEmailArrFromUserUidArr' and 'getUserUidArrFromUserEmailArr'. Both access the corresponding servlet and returns its response. On servlet error, returns an empty array instead.

### Check lists (check `x` in `[ ]` of list items)

- [  ] Test written/updating
- [  ] Tests passing
- [x] Coding style (indentation, etc)

Testing is addressed in #99.

### CODE REVIEW

File: frontend/src/components/AuthUtils/index.js [hunk 1]
```diff
@@ -71,12 +71,74 @@ export function signOut() {
...
+}
+
+/**
+ * Retrieves the user UIDs corresponding to an array of user emails.
+ *
+ * The returned array of user UIDs are sorted in alphabetical order.
+ *
+ * @param {Array<String>} emailArr Array of user emails for which we wish to
+ * retrieve the corresponding user UIDs.
+ * @return {Promise<Array<String>>} The array of user UIDs corresponding to
```
Comment (ananyeet) [inline frontend/src/components/AuthUtils/index.js:111]:
```suggestion
 * @param {Array.<String>} emailArr Array of user emails for which we wish to
 * retrieve the corresponding user UIDs.
 * @return {Promise<Array.<String>>} The array of user UIDs corresponding to
```
jsdoc formatting

File: frontend/src/components/AuthUtils/index.js [hunk 2]
```diff
@@ -71,12 +71,74 @@ export function signOut() {
...
 
+/**
+ * Retrieves the user emails corresponding to an array of user UIDs.
+ *
+ * The returned array of user emails are sorted in alphabetical order.
+ *
+ * @param {Array<String>} uidArr Array of user UIDs for which we wish to
+ * retrieve the corresponding user emails.
+ * @return {Promise<Array<String>>} The array of user emails corresponding to
+ * the given UIDs.
```
Comment (ananyeet) [inline frontend/src/components/AuthUtils/index.js:82]:
```suggestion
 * @param {Array.<String>} uidArr Array of user UIDs for which we wish to
 * retrieve the corresponding user emails.
 * @return {Promise<Array.<String>>} The array of user emails corresponding to
 * the given UIDs.
```
jsdoc formatting

File: frontend/src/components/AuthUtils/index.js [hunk 3]
```diff
@@ -71,12 +71,74 @@ export function signOut() {
...
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(emailArr)
+  };
+
+  try {
+    const response = await fetch(fetchUrl, fetchOptions);
+    const text = await response.text();
+    return JSON.parse(text);
```
Comment (ananyeet) [inline frontend/src/components/AuthUtils/index.js:127]: What's coming out of the json here?

Comment (keiffer01) [reply frontend/src/components/AuthUtils/index.js:127]: The servlet returns an array of the converted data (UIDs -> email or email -> UIDs) as a JSON string.

File: frontend/src/components/AuthUtils/index.js (hunk 1 above)
Comment (keiffer01) [reply frontend/src/components/AuthUtils/index.js]: Fixed.

File: frontend/src/components/AuthUtils/index.js (hunk 2 above)
Comment (keiffer01) [reply frontend/src/components/AuthUtils/index.js]: Fixed.

**Comment by keiffer01** [comment]
> What happens if, on this site, I manually craft a post request to these servlet functions with a list of emails or uids? Will I get the list back, or is there some form of security preventing arbitrary post requests from being responded to? This would be interesting to test using something like postman: https://learning.postman.com/docs/sending-requests/requests/

Unfortunately I think you _can_ just craft a POST request from anywhere like this, though it might be possible to set rules in the Google Cloud console to have the backend component only accept requests from the frontend component. It's something that I'll try to look into.

**Pull request merged**